core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/Utils.kt

```diff
@@ -337,28 +337,110 @@ internal fun <T : Comparable<T>> T.between(left: T, right: T, includeBoundaries:
         this > left && this < right
     }
 
-private const val DELIMITERS = "[_\\s]"
-public val DELIMITERS_REGEX: Regex = DELIMITERS.toRegex()
-public val DELIMITED_STRING_REGEX: Regex = ".+$DELIMITERS.+".toRegex()
+// Single regex to split words by non-alphanumeric characters, camelCase, and numbers
 
-internal val CAMEL_REGEX = "(?<=[a-zA-Z])[A-Z]".toRegex()
+internal val CAMEL_DEFAULT_DELIMITERS_REGEX =
+    "[^a-zA-Z0-9]+|(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])|(?<=\\d)(?=[a-zA-Z])|(?<=[a-zA-Z])(?=\\d)".toRegex()
```

Comment:
Hmm, I'm wondering whether "a-zA-Z" correctly catches all letters. There are more letters than just a-z; there are letters like óäàãæ, not to mention other alphabets. Best to use the "\p" notation in Regex to catch all cases. I used something similar in KoDEx to recreate the Kotlin spec for an "identifier" in Regex: https://github.com/Jolanrensen/KoDEx/blob/97abf961186d2bb6c75756e9ec1946dd853e721e/kodex-common/src/main/kotlin/nl/jolanrensen/kodex/docContent/DocContent.kt#L319 There are even \p notations for uppercase and lowercase letters I believe.